js/main.js

```diff
@@ -33,4 +36,46 @@ function getRandomNumber(collectionLength) {
   return integer;
 }
 
+function handleClick(event) {
+  if (event.target.matches('.fa-solid')) {
+    var collectionData = {
+      collectionName: data.NFTvisible[0].collectionName,
+      collectionPhoto: null,
+      likes: null,
+      dislikes: null
+    };
+
+    var singleNFTData = {
+      name: data.NFTvisible[0].name,
+      data: data.NFTvisible[0].image
+    };
+
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', 'https://api.opensea.io/api/v1/collection/' + collectionData.collectionName.toLowerCase());
+    xhr.responseType = 'json';
+    xhr.addEventListener('load', function () {
+      collectionData.collectionPhoto = xhr.response.collection.image_url;
+    });
+    xhr.send();
+
+    if (Object.entries(data.ratingsInfo).length === 0) {
+      data.ratingsInfo[collectionData.collectionName] = collectionData;
+    }
+
+    for (var key in data.ratingsInfo) {
+      if (key !== data.NFTvisible[0].collectionName) {
+        data.ratingsInfo.push(collectionData);
+      } else if (event.target.matches('.fa-heart')) {
+        data.ratingsInfo[key].likes++;
+      } else if (event.target.matches('.fa-star')) {
+        data.superliked.push(singleNFTData);
+      } else if (event.target.matches('.fa-thumbs-down')) {
+        data.ratingsInfo[key].dislikes++;
+      }
```

Comment:
This doesn't need to be done in a loop, you can just use `data.ratingsInfo[collectionData.collectionName]`